test/parse.js

```diff
@@ -0,0 +1,433 @@
+					expect: '{"spaceId":"a98rgb","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(a98-rgb 0 100% 50%)",
+					expect: '{"spaceId":"a98rgb","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(prophoto-rgb 0 1 .5)",
+					expect: '{"spaceId":"prophoto","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(prophoto-rgb 0 100% 50%)",
+					expect: '{"spaceId":"prophoto","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(acescc 0 1 .5)",
+					expect: '{"spaceId":"acescc","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(acescc 0 100% 50%)",
+					expect: '{"spaceId":"acescc","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(acescg 0 1 .5)",
+					expect: '{"spaceId":"acescg","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(acescg 0 100% 50%)",
+					expect: '{"spaceId":"acescg","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(xyz 0 1 .5)",
+					expect: '{"spaceId":"xyz-d65","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(xyz 0 100% 50%)",
+					expect: '{"spaceId":"xyz-d65","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(xyz-d65 0 1 .5)",
+					expect: '{"spaceId":"xyz-d65","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(xyz-d65 0 100% 50%)",
+					expect: '{"spaceId":"xyz-d65","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(xyz-d50 0 1 .5)",
+					expect: '{"spaceId":"xyz-d50","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(xyz-d50 0 100% 50%)",
+					expect: '{"spaceId":"xyz-d50","coords":[0,1,0.5],"alpha":1}'
+				},
+				{
+					args: "color(jzazbz 0 25% -50%)",
+					expect: '{"spaceId":"jzazbz","coords":[0,0.125,-0.25],"alpha":1}'
+				},
+				{
+					args: "color(--jzazbz 0 25% -50%)",
+					expect: '{"spaceId":"jzazbz","coords":[0,0.125,-0.25],"alpha":1}'
+				},
+				{
+					args: "color(--hct 0.25turn 50% 25)",
+					expect: '{"spaceId":"hct","coords":[90,72.5,25],"alpha":1}'
+				},
+				{
+					name: "With transparency",
+					args: "color(display-p3 0 1 0 / .5)",
+					expect: '{"spaceId":"p3","coords":[0,1,0],"alpha":0.5}'
+				},
+				{
+					name: "No arguments",
+					args: "color(display-p3)",
+					expect: '{"spaceId":"p3","coords":[0,0,0],"alpha":1}'
+				},
+				{
+					name: "No arguments / alpha",
+					args: "color(display-p3 / .5)",
+					expect: '{"spaceId":"p3","coords":[0,0,0],"alpha":0.5}'
+				},
+				{
+					name: "Fewer arguments",
+					args: "color(display-p3 1)",
+					expect: '{"spaceId":"p3","coords":[1,0,0],"alpha":1}'
+				},
+				{
+					name: "Fewer arguments / alpha",
+					args: "color(display-p3 1 / .5)",
+					expect: '{"spaceId":"p3","coords":[1,0,0],"alpha":0.5}'
+				},
+				{
+					name: "none red",
+					args: "color(display-p3 none 1 .5)",
+					expect: '{"spaceId":"p3","coords":[null,1,0.5],"alpha":1}'
+				}
+			]
+		},
+		{
+			name: "hsl()",
+			tests: [
+				{
+					name: "hsl(), commas",
```

Comment:
Does the output not show the parent name before the child name or in a way that implies containment?

Reply:
It does, the function portion of the name is redundant